### PR TITLE
feat: add programmatic scrolling to scroller

### LINKED ISSUE #6071: Scroller: add support for scrolling to the top
### Describe your motivation

I have a wizard component where the Scroller component hosts the individual wizard pages. I'm calling Scroller.setContent() as the user progresses through the wizard. Unfortunately, the scroller keeps its scrolling position even after a new content is set, causing the new wizard pane to be scrolled downwards.

### Describe the solution you'd like

Either the Scroller should automatically scroll to the top upon content change, or there could be a function to do that.

### Describe alternatives you've considered

_No response_

### Additional context

Vaadin 24.3.3.

### PR DESCRIPTION
**Pull request by FrediWa**
## Description

Add an API to allow the user to scroll a Scroller to top and bottom.

Fixes #6071 

## Type of change

- [X] Feature

### CODE REVIEW

File: vaadin-ordered-layout-flow-parent/vaadin-ordered-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/orderedlayout/it/ScrollerPage.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.vaadin.flow.router.Route;
+
+/**
+ * View for the ordered layouts {@link HorizontalLayout}, {@link VerticalLayout}
+ * and {@link Scroller}.
+ *
+ * @author Vaadin Ltd
+ */
+@Route("vaadin-ordered-layout/scroller")
+public class ScrollerPage extends Div {
```
Comment (tomivirkki) [inline vaadin-ordered-layout-flow-parent/vaadin-ordered-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/orderedlayout/it/ScrollerPage.java:32]: The scroller on the test page is not scrollable

Screenshot 2024-07-24 at 11 00 13

File: vaadin-ordered-layout-flow-parent/vaadin-ordered-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/orderedlayout/it/ScrollerPage.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.Scroller;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.router.Route;
+
+/**
+ * View for the ordered layouts {@link HorizontalLayout}, {@link VerticalLayout}
+ * and {@link Scroller}.
```
Comment (tomivirkki) [inline vaadin-ordered-layout-flow-parent/vaadin-ordered-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/orderedlayout/it/ScrollerPage.java:27]: This test page is only for the scroller

File: vaadin-ordered-layout-flow-parent/vaadin-ordered-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/orderedlayout/it/ScrollerPage.java (hunk 1 above)
Comment (FrediWa) [reply vaadin-ordered-layout-flow-parent/vaadin-ordered-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/orderedlayout/it/ScrollerPage.java:29]: I mistakenly removed the width and height from the view while cleaning the file.

I now added a test that makes sure that when the content is at the bottom, it's not at the top. This assert shouldn't hold if the test is scrollable.

File: vaadin-ordered-layout-flow-parent/vaadin-ordered-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/orderedlayout/it/ScrollerPage.java (hunk 2 above)
Comment (FrediWa) [reply vaadin-ordered-layout-flow-parent/vaadin-ordered-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/orderedlayout/it/ScrollerPage.java]: Fixed!

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**